### PR TITLE
feat: support new upload-to-job-server publsih

### PR DESCRIPTION
**Pull request by bloodearnest**
- uploads a zip of approved files to job-server.
 - uses urllib rather than requests to avoid the dependency
 - `-n` switch to use, still supports github method
 - requires backend token for auth

### CODE REVIEW

File: publisher/upload.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    release_hash = hash_files(release_dir, files)
+    workspace = manifest["workspace"]
+
+    zip_buffer = io.BytesIO()
+    with ZipFile(zip_buffer, "w") as zip_file:
+        for path in files:
+            zip_file.write(release_dir / path, arcname=str(path))
+
+    data = zip_buffer.getvalue()
+    request = Request(
```
Comment (evansd) [inline publisher/upload.py:39]: Nice. It's always annoyed me that so many little scripts I write have `requests` as their only non-stdlib dependency. But I've always had this vague fear that doing HTTP stuff without it was gnarly and complicated. Turns out not to be too bad!

**Review by evansd — APPROVED** [review]
Looks good!

File: publisher/release.py [hunk 2]
```diff
@@ -143,7 +143,7 @@ def main(study_repo_url, token, files):
                             "--set-upstream",
                             "origin",
                             release_branch,
-                            ]
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply publisher/release.py:146]: Yeah, I ended up doing a fair bit of retroactive git surgery on this branch, will try keep it cleaner in future

File: publisher/upload.py (hunk 1 above)
Comment (bloodearnest) [reply publisher/upload.py:39]: Yeah, basic use has got a lot better. The testing was a PITA, though

**Pull request merged**